entry

```diff
@@ -1,5 +1,81 @@
 #!/bin/bash
 
+helm_update() {
+  if [ "$HELM" == "helm_v3" ]; then
+    LINE="$($HELM ls --all -f "^$NAME\$" --output json | jq -r "$JQ_CMD" | tr '[:upper:]' '[:lower:]')"
+  else
+    LINE="$($HELM ls --all "^$NAME\$" --output json | jq -r "$JQ_CMD" | tr '[:upper:]' '[:lower:]')"
+  fi
+  INSTALLED_VERSION=$(echo $LINE | cut -f1 -d,)
+  STATUS=$(echo $LINE | cut -f2 -d,)
+
+  if [ -e /config/values.yaml ]; then
+      VALUES="--values /config/values.yaml"
+  fi
+
+  if [ "$1" = "delete" ]; then
+      if [ -z "$INSTALLED_VERSION" ]; then
+          exit
+      fi
+      if [ "$HELM" == "helm_v3" ]; then
+        $HELM uninstall $NAME
+      else
+        $HELM delete $NAME || true
+        $HELM "$@" --purge $NAME
+      fi
+      exit
+  fi
+
+  if [ -z "$INSTALLED_VERSION" ] && [ -z "$STATUS" ]; then
+      $HELM "$@" $NAME $CHART $VALUES
+      exit
+  fi
+  if [ -z "$VERSION" ] || [ "$INSTALLED_VERSION" = "$VERSION" ]; then
+      if [ "$STATUS" = "deployed" ]; then
+          echo Already installed $NAME, upgrading
+          $HELM upgrade $NAME $CHART $VALUES
+          exit
+      fi
+  fi
+
+  if [ "$STATUS" = "failed" ] || [ "$STATUS" = "deleted" ]; then
+    if [ "$HELM" == "helm_v3"]; then
+        $HELM uninstall $NAME
+    else
+        $HELM "$@" --purge $NAME
+    fi
+    echo Deleted
+    $HELM "$@" $NAME_ARG $NAME $CHART $VALUES
+    exit
+  fi
+
+  $HELM "$@" $NAME_ARG $NAME $CHART $VALUES
+}
+
+helm_repo_init() {
+  # if the chart is url skip repo update
+  if grep -q -e "https\?://" <<< "$CHART"; then
+    echo "chart path is a url, skipping repo update"
+    return
+  fi
+
+  if [ "$HELM" == "helm_v3" ]; then
+    $HELM repo add stable https://kubernetes-charts.storage.googleapis.com/
```

Comment:
The stable repo is effectively deprecated with v3.  Can we check that $REPO starts with stable/ and then only add the stable repo?  For helm 2, keep the existing logic, this is just for helm 3.

Reply:
sure will do